### ISSUE TITLE
[Feature request] Offline Mode

### ISSUE THREAD

**Issue by kuflierl** [leinelissen/jellyfin-audio-player#160]
Personally it would be a great idea to implement an offline mode where stuff not downloaded is grayed out and network requests aren't triggered in the background. It could automatically trigger when connection to the server is not possible or a small toggle is toggled. This would reduce battery usage, make offline playing easier and make the fact that that the player is offline clearer to the user. It could also be used when someone doesn't have much data left on their data cap but would still like to keep data on for other apps.

**Comment by leinelissen** [comment]
Hey, thanks for the input! I wanted to get your thoughts on how this should work out. Currently, no network requests (other than images) are made if a song is downloaded. There is an icon next to each track that indicates whether it is downloaded or streamed directly. Would the purpose of this mode be to prevent accidentally playing back tracks that haven't been downloaded? I would prefer to follow OS settings for being offline unless there is a reason for deviating from it.

**Comment by wildrun0** [comment]
I'm not a topic starter, but I have some thoughts. If there is no internet access, albums are displayed from the cache, but there are no songs in them. It takes some time to find the album I downloaded. Perhaps "downloaded" menu should be done in other way, where all downloaded albums are presented, and not just songs that you can't even play

**Comment by kuflierl** [comment]
> I'm not a topic starter, but I have some thoughts. If there is no internet access, albums are displayed from the cache, but there are no songs in them. It takes some time to find the album I downloaded. Perhaps "downloaded" menu should be done in other way, where all downloaded albums are presented, and not just songs that you can't even play

Yeah, that's why i suggested graying out albums not yet downloaded in offline mode or at least add a small badge on them to signify that they are fully or partially downloaded. I think making the downloaded menu clickable would also go a huge way. Great suggestion!

**Comment by leinelissen** [comment]
Making the download tab more navigable sounds like a great solution. It does require some more UX work around removing things and separating out the playlists and albums again. I'll see what I can do.

**Comment by mattcoxonline** [comment]
Here's how I use the app and I hope it gives you an idea of how I would use this feature:

- I have a local Jellyfin server set up. It works in my home on a local IP.
- When I am out of the house, I cannot access my library.
- I would expect to 'Download' albums at home and be able to play Downloaded albums when out the house.

To add to the suggestions above; this means I would like the app to either automatically realise there is no current connection to the Jellyfin server and only show Downloaded items. Or, to be able to manually toggle to an Offline Mode which only shows downloaded albums.

I'm an Android user, but I think AmpFin on iOS does something similar to this: https://github.com/rasmuslos/AmpFin
